c7n/reports/csvout.py

```diff
@@ -263,44 +266,56 @@ def record_set(session_factory, bucket, key_prefix, start_date):
     """
 
     s3 = local_session(session_factory).client('s3')
-    marker = key_prefix.strip("/") + "/" + start_date.strftime(
-        '%Y-%m-%d-00') + "/resources.json.gz"
+
+    today = date.today()
+    current_date = start_date.date()
+    date_folders = []
```

Comment:
I was thinking that, but will that work in the nested structure at the month boundaries?

If my starting point is 2016/04/02 and I have 3 days of history, will it properly go and get 2016/04/02, 2016/04/01 and then back up to 2016/03/31?